src/types/countly.d.ts

```diff
@@ -20,7 +20,8 @@ declare module 'countly-sdk-web' {
 
   export type eventTypes = 'apm' | 'attribution' | 'clicks' | 'crashes' | 'events' | 'feedback' | 'forms' |
   'location' | 'scrolls' | 'sessions' | 'star-rating' | 'users' | 'views'
-  export type consentTypes = 'all' | 'minimal' | 'marketing' | 'tracking' | 'performance'
+  export type consentTypesExceptAll = 'minimal' | 'marketing' | 'tracking' | 'performance'
+  export type consentTypes = 'all' | consentTypesExceptAll
```

Comment:
nit: I still don’t think we need consentTypes explicitly typed, even for consuming the lib (only our MetricsProvider functions need it, and those can use the type of the keys defined from groupedFeatures), but this is an improvement for sure